contentscript.js

```diff
@@ -1,302 +0,0 @@
-                middleAndLastNameCombos[middleAndLastNameCombosIndex]) {
-                middleNames = middleAndLastNameCombos[middleAndLastNameCombosIndex];
-                middleAndLastNameCombosIndex++;
-                tryMiddleAndLastNameCombos = middleAndLastNameCombos[middleAndLastNameCombosIndex];
-                if (!tryMiddleAndLastNameCombos) {
-                    restoreMiddleNames = true;
-                }
-                GetProfessorRating(element, fullName, lastName, originalLastName, firstName, originalFirstName, 
-                    middleNames, originalMiddleNames, tryNicknames, nicknamesIndex, tryMiddleAndLastNameCombos,
-                    middleAndLastNameCombosIndex, tryMiddleNameAsFirst, tryMiddleNames);
-            }
-            // Try again with nicknames for the first name
-            else if (tryNicknames && nicknames[originalFirstName]) {
-                firstName = nicknames[originalFirstName][nicknamesIndex];
-                nicknamesIndex++;
-                tryNicknames = nicknames[originalFirstName][nicknamesIndex];
-                if (!tryNicknames) {
-                    restoreFirstName = true;
-                }
-                GetProfessorRating(element, fullName, lastName, originalLastName, firstName, originalFirstName, 
-                    middleNames, originalMiddleNames, tryNicknames, nicknamesIndex, tryMiddleAndLastNameCombos,
-                    middleAndLastNameCombosIndex, tryMiddleNameAsFirst, tryMiddleNames);
-            }
-            // Try again with the middle name as the first name
-            else if (tryMiddleNameAsFirst && !tryMiddleNames && originalMiddleNames.length > 0) {
-                tryMiddleNameAsFirst = false;
-                restoreFirstName = true;
-                GetProfessorRating(element, fullName, lastName, originalLastName, firstName, originalFirstName, 
-                    middleNames, originalMiddleNames, tryNicknames, nicknamesIndex, tryMiddleAndLastNameCombos,
-                    middleAndLastNameCombosIndex, tryMiddleNameAsFirst, tryMiddleNames);
-            }
-            // Try again with middle names
-            else if (!tryMiddleNames && originalMiddleNames.length > 0) {
-                tryMiddleNames = true;
-                GetProfessorRating(element, fullName, lastName, originalLastName, firstName, originalFirstName, 
-                    middleNames, originalMiddleNames, tryNicknames, nicknamesIndex, tryMiddleAndLastNameCombos,
-                    middleAndLastNameCombosIndex, tryMiddleNameAsFirst, tryMiddleNames);
-            }
-            // Set link to search results if not found
-            else {
-                element.textContent = `${element.textContent} (NF)`;
-                element.setAttribute('href', 
-                `https://www.ratemyprofessors.com/search/teachers?query=${originalLastName}&sid=${schoolId}`);
-            }
-        }        
-    });
-}
```

Comment:
can we restore the logic i had here for searching names?

Reply:
I could, but you might have to help me understand what logic you were going for and i untangle it from the old RMP API queries since the new api has a different structure/everything can be done with basically one query now.


What im kinda going for with the new code is like a two-stage approach of:
1. Generate a list of search strings
2. Try them out one by one until you get results

Do you think you could do the first step of creating a function that takes in something representing a professors name and generates a list of search strings to try in order of likelihood?

I can provide basically any value that exists in the first part of the `arrive` callback (i.e. raw name from the page, filtered name, name as a list of strings per part, etc) and i also have a list of nearly every professors name at RIT that i scraped from ScheduleMaker if you want that for testing.

This should make it easier to change the api code in future if needed without affecting the name searching process

Reply:
Can we simplify the logic to this?

Given a [First Name] [First Middle Name] [Second Middle Name] [Last Name]

[First Name] [Last Name]
[try all nicknames for First Name] [Last Name]
[First Middle Name] [Last Name]
[try all nicknames for First Middle Name] [Last Name]
[Second Middle Name] [Last Name]
[try all nicknames for Second Middle Name] [Last Name]

Reply:
Oh ok, that makes sense, ill see if i can write that up into code when i get a chance